app/views/shared/_order.html.erb

```diff
@@ -244,10 +244,9 @@
   <div class="row order-data">
     <br/><br/>
     <h2 class="pull-left">Items for Delivery</h2>
-    <% if !current_user.buyer_only? && (order.delivery_status == 'pending' || order.delivery_status == 'partially delivered') %>
+    <% if !current_user.buyer_only? && (current_user.can_manage_market?(order.market) || order.delivery.delivery_schedule.direct_to_customer?) && (order.delivery_status == 'pending' || order.delivery_status == 'partially delivered') %>
       <%= button_tag "Mark all delivered", class: 'pull-right btn btn--small btn--primary mobile-block', style: 'margin-left: 15px;', id: 'mark-all-delivered', type: 'button' %>
-    <% end %>
-    <% if
+    <% elsif
         (current_market.is_buysell_market? && current_user.can_manage_market?(order.market) && !order.undelivered_for_user?(current_user)) ||
         (current_market.is_consignment_market? && current_user.can_manage_market?(order.market) && (order.delivery_status == 'delivered' || order.delivery_status == 'partially delivered') && Inventory::Utils.consignment_can_undeliver?(order))
     %>
```

Comment:
@LocalOrbit/developers where is the best place to move logic like this...a) in a regular template but also b) like in this case of a in a shared template? Should the controller be responsible for this? OrderDecorator? Some composite abstract controller method?